### PR TITLE
[Component with its own class name as service type] LPS-196642 Refactor components from commerce, objects and lima modules 

### PR DESCRIPTION
**Pull request by MrJohn1911**
Ticket: https://liferay.atlassian.net/browse/LPS-196642

### CODE REVIEW

File: modules/apps/commerce/commerce-frontend-impl/src/main/java/com/liferay/commerce/frontend/internal/cart/CommerceCartResource.java [hunk 1]
```diff
@@ -304,10 +327,327 @@ private Response _getResponse(Object object) {
...
+
 	@Reference
 	private Portal _portal;
 
+	@Reference
+	private ProductHelper _productHelper;
+
+	private class CommerceCartResourceUtil {
+
+		public Cart getCart(
```
Comment (rafaprax) [inline modules/apps/commerce/commerce-frontend-impl/src/main/java/com/liferay/commerce/frontend/internal/cart/CommerceCartResource.java:356]: just move the methods from CommerceCartResourceUtil to the outer class. No need to have an inner class here.

File: modules/apps/adaptive-media/adaptive-media-image-service/src/main/java/com/liferay/adaptive/media/image/service/impl/AMImageEntryLocalServiceImpl.java [hunk 2]
```diff
@@ -302,6 +304,74 @@ public boolean hasAMImageEntryContent(
 		return _imageStorage.hasContent(fileVersion, configurationUuid);
 	}
 
+	public class ImageStorage {
```
Comment (rafaprax) [inline modules/apps/adaptive-media/adaptive-media-image-service/src/main/java/com/liferay/adaptive/media/image/service/impl/AMImageEntryLocalServiceImpl.java:307]: keep it an isolated class and its dependencies as constructor parameters.

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by MrJohn1911** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-type-virtual-order-content-web/src/main/java/com/liferay/commerce/product/type/virtual/order/content/web/internal/security/resource/permission/CommerceVirtualOrderItemPermissionUtil.java [hunk 3]
```diff
@@ -10,28 +10,21 @@
...
-
 /**
  * @author Alessio Antonio Rendina
  */
-@Component(service = CommerceVirtualOrderItemPermission.class)
-public class CommerceVirtualOrderItemPermission {
+public class CommerceVirtualOrderItemPermissionUtil {
 
-	public boolean contains(
+	public static boolean contains(
```
Comment (rafaprax) [inline modules/apps/commerce/commerce-product-type-virtual-order-content-web/src/main/java/com/liferay/commerce/product/type/virtual/order/content/web/internal/security/resource/permission/CommerceVirtualOrderItemPermissionUtil.java:18]: remove this classe and call the `commerceVirtualOrderItemModelResourcePermission#contains` directly.

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Pull request closed without merging**